selftests/unit/test_test.py

```diff
@@ -21,10 +21,15 @@
 """
 
 
-class DummyTest(test.Test):
+def get_tst():
```

Comment:
These "fake" test classes can be hidden from unittest discovery by taking them out of the main module level, as you did with `get_tst()`.  But, I think by putting those class definitions inside the real unittest classes that use them gives a much better readability.  Example:

```diff
--- a/selftests/unit/test_test.py
+++ b/selftests/unit/test_test.py
@@ -20,19 +20,11 @@ FAIL_SCRIPT_CONTENTS = """#!/bin/sh
 false
 """
 
+class TestClassTestUnit(unittest.TestCase):
 
-def get_tst():
-    """
-    To avoid detecting this class as unittest let's return it on demand
-    """
     class DummyTest(test.Test):
-
         def test(self):
             pass
-    return DummyTest
-
-
-class TestClassTestUnit(unittest.TestCase):
 
     def setUp(self):
         self.tmpdir = tempfile.mkdtemp(prefix="avocado_" + __name__)
@@ -44,8 +36,8 @@ class TestClassTestUnit(unittest.TestCase):
     def testUglyName(self):
         def run(name, path_name):
             """ Initialize test and check the dirs were created """
-            tst = get_tst()("test", test.TestName(1, name),
-                            base_logdir=self.tmpdir)
+            tst = self.DummyTest("test", test.TestName(1, name),
+                                 base_logdir=self.tmpdir)
             self.assertEqual(os.path.basename(tst.logdir), path_name)
             self.assertTrue(os.path.exists(tst.logdir))
             self.assertEqual(os.path.dirname(os.path.dirname(tst.logdir)),
@@ -69,8 +61,8 @@ class TestClassTestUnit(unittest.TestCase):
 
     def testLongName(self):
         def check(uid, name, variant, exp_logdir):
-            tst = get_tst()("test", test.TestName(uid, name, variant),
-                            base_logdir=self.tmpdir)
+            tst = self.DummyTest("test", test.TestName(uid, name, variant),
+                                 base_logdir=self.tmpdir)
             self.assertEqual(os.path.basename(tst.logdir), exp_logdir)
             return tst
 
@@ -106,22 +98,21 @@ class TestClassTestUnit(unittest.TestCase):
     def testAllDirsExistsNoHang(self):
         flexmock(os.path)
         os.path.should_receive('exists').and_return(True)
-        self.assertRaises(exceptions.TestSetupFail, get_tst(), "test",
+        self.assertRaises(exceptions.TestSetupFail, self.DummyTest, "test",
                           test.TestName(1, "name"), base_logdir=self.tmpdir)
 
 
 class TestClassTest(unittest.TestCase):
 
-    def setUp(self):
-        class AvocadoPass(test.Test):
-
-            def test(self):
-                variable = True
-                self.assertTrue(variable)
-                self.whiteboard = 'foo'
+    class AvocadoPass(test.Test):
+        def test(self):
+            variable = True
+            self.assertTrue(variable)
+            self.whiteboard = 'foo'
 
+    def setUp(self):
         self.base_logdir = tempfile.mkdtemp(prefix='avocado_' + __name__)
-        self.tst_instance_pass = AvocadoPass(base_logdir=self.base_logdir)
+        self.tst_instance_pass = self.AvocadoPass(base_logdir=self.base_logdir)
         self.tst_instance_pass.run_avocado()
 
     def testClassAttributesName(self):
```